Update button docs and toggle button (#1827)
grommet/hpe-design-system: aries-site/src/pages/components/button.mdx

- Use buttons when you are performing an action, such as: "submit," "merge," "create new," "upload," etc.
- An action almost always occurs on the same page

### Toggle Buttons

> Toggle buttons are used either to switch between two states, such as check and unchecked, or to trigger two actions like on or off.

Reviewer: ```suggestion
Toggle buttons are used either to switch between two states, such as checked and unchecked, or to trigger two actions like on or off.
```

#### Toggle Buttons with Icons

> Icon buttons have an aria-label that will describe the action that the user will be performing when using screen readers.

Reviewer 2: For consistency with the text below:

 "... have an `a11y-title`, `aria-label`, or button `label` that describes the action ..."

**Best Practices:**
- Limit to 2-3 descriptive words.

> - Use verbs first to describe the action such as “Toggle” , “Favorite” or “Save”.

Reviewer 2: The code in ThemeModeToggle uses the verb "Switch".  Maybe that should be mentioned in this list.

In the context of toggle buttons with icons, "Favorite" and "Save" as verbs don't fit here.  It needs the two toggle states with associated icons. E.g. "Like/Dislike", "Light/Dark", "Hide/View".

- Use sentence case capitalization.

> - Words such as “the”, “a” be avoided.

Reviewer: ```suggestion
- Words such as “the” and “a” should be avoided.
```

For Buttons with Icons ONLY, it is important that you show the two following elements:

1. Show the current state of the button through the use of an icon that best represents the action.

> 2. Inform the user of what is going to happen next by including `a11yTitle`, `aria-label`, or changing the button's `label`.

Reviewer 2: Drop "that best represents the action", as it is a motherhood-and-apple-pie statement and offers no specific guidance.  It's a "Do your best" gee-up.  ie. every UI designer/developer will be making a personal choice about what best represents the action.

Drop ", or changing the buttons's `label`", since there is no label for "Buttons with Icons ONLY".  However, this needs to be captured in the section above. (I'll add a comment to that too).

I'll propose the following re-wording...

For Buttons with Icons ONLY, the icon shows current state and the aria-label describes the next state.  It is important that you show the two following elements:

1. Show the current state of the button through the use of an icon
2. Inform the user of what is going to happen next by including `a11yTitle` or `aria-label`.

> 2. Inform the user of what is going to happen next by including `a11yTitle` or `aria-label`.

Reviewer: ```suggestion
2. Inform the user of what will happen when the button is clicked by including `a11yTitle` or `aria-label`.
```

- Use buttons when you are performing an action, such as: "submit," "merge," "create new," "upload," etc.
- An action almost always occurs on the same page

### Toggle Buttons

> Toggle buttons are buttons with either two states: check and unchecked or used to trigger two actions like on or off.

Reviewer: ```suggestion
Toggle buttons are used either to switch between two states, such as check and unchecked, or to trigger two actions like on or off.
```

#### Toggle Buttons with Icons

Icon buttons have an aria-label that will describe the action that the user will be performing when using screen readers.

**Best Practices:**

> - Don’t use more than 2-3 words
> - Use verbs first to describe the action such as “Toggle” , “Favorite” or “Save”
> - Use sentece case capitalization
> - It is recommended that words such as “the”, “a” be avoided

Reviewer: ```suggestion
- Limit to 2-3 descriptive words.
- Use verbs first to describe the action such as “Toggle” , “Favorite” or “Save”.
- Use sentence case capitalization.
- Words such as “the”, “a” be avoided.
```

For Buttons with Icons ONLY, it is important that you show the two following elements:

> 1. Show the current state of the button through the use of an icon that best represents the action

Reviewer: ```suggestion
1. Show the current state of the button through the use of an icon that best represents the action.
```

Reviewer: It would be helpful to supplement this with an example such as show/hide password, or lock / unlock editing mode, or even the play/pause.

Author: I agree. I've added a ticket in our backlog to add this example along with some guidance. Ticket #1840

> 2. Inform the user of what is going to happen next by including a11y text and or changing the label

Reviewer: ```suggestion
2. Inform the user of what is going to happen next by including `a11yTitle`, `aria-label`, or changing the button's `label`.
```

#### Where to use Toggle Buttons

> - Add checkboxes in forms

Reviewer: This is confusing / unclear to me.

Author: I agree. I'm going to come back to this section

Author: We took this section out for now since we have another ticket where we will clarify this more. Ticket #1657

- Play/pause a video
- Like/dislike a comment
- Favorite/unfavorite an image

> - To show the visibility of an asset or a collection such as that in an accordion

Reviewer: This is confusing / unclear to me.

Author: We took this section out for now since we have another ticket where we will clarify this more. Ticket #1657

> - To select an option among many options like in a toggle button

Reviewer: This is confusing / unclear to me.

Author: Will come back tot his section with more clear content

Author: We took this section out for now since we have another ticket where we will clarify this more. Ticket #1657
